kratos/includes/registry_item.h

```diff
@@ -97,18 +223,30 @@ class RegistryItem
         std::string const& ItemName,
         TArgumentsList&&... Arguments)
     {
-        KRATOS_ERROR_IF(this->HasItem(ItemName)) << "The RegistryItem " << this->Name() << " already has an item with name " << ItemName << std::endl;
+        KRATOS_ERROR_IF(this->HasItem(ItemName)) << "The RegistryItem '" << this->Name() << "' already has an item with name " << ItemName << "." << std::endl;
+        KRATOS_ERROR_IF(this->HasValue()) <<
+            "Trying to add '"<< ItemName << "' item to the RegistryItem '" << this->Name() << "' but this already has value. Items cannot have both value and subitem." << std::endl;
         auto insert_result = mSubRegistryItem.emplace(std::make_pair(ItemName, Kratos::make_unique<TItemType>(ItemName, std::forward<TArgumentsList>(Arguments)...)));
-        KRATOS_ERROR_IF_NOT(insert_result.second) << "Error in inserting " << ItemName << " in registry item with name " << this->Name() << std::endl;
+        KRATOS_ERROR_IF_NOT(insert_result.second) << "Error in inserting '" << ItemName << "' in registry item with name '" << this->Name() << "'." << std::endl;
```

Comment:
Not exactly. We added the `HasValue` check because, even though it's technically possible, we don't want items to have both value and subitem. The one you suggest to eliminate checks that the insertion (of either a value or a subitem) is correct.

Reply:
My bad, I meant it's the same as the `HasItem` check. Both fail iff `mSubRegistryItem already has an entry with ItemName key.

Reply:
Ah! Agree. I think we can safely remove it.